### PR TITLE
Upgrade pymongo from v3 to v4

### PR DESCRIPTION
**Pull request by sengineer0**
ref: https://github.com/biothings/biothings.api/issues/222

### CODE REVIEW

**Comment by sengineer0** [comment]
Currently got stuck with the error: `pymongo.errors.AutoReconnect connection pool paused`

Selection_001

**Comment by sengineer0** [comment]
> Currently got stuck with the error: `pymongo.errors.AutoReconnect connection pool paused`
> 
> Selection_001

this problem is fixed

File: biothings/utils/configuration.py [hunk 1]
```diff
@@ -245,9 +246,21 @@ def get_value_from_db(self, name):
...
+                    raise AttributeError(name)
+                retry += 1
+                break
+            except AutoReconnect:
+                self._db.__database.close()
+                self._db = self._get_db_function()
+
         if not doc:
-            raise AttributeError(name)
+            raise Exception("Cannot find document due to connection problem")
```
Comment (newgene) [inline biothings/utils/configuration.py]: @sengineer0 any more specific exception can be raised here?

Comment (sengineer0) [reply biothings/utils/configuration.py]: you meant we should define a custom Exception for this exception, like MaxRetryAutoReconnectException, do you?

Comment (newgene) [reply biothings/utils/configuration.py]: @sengineer0 Maybe first to check if there is a suitable exception from pymongo or Python's built-in exceptions. If not, yes, create a new exception is better than just use the generic Exception class.

Comment (sengineer0) [reply biothings/utils/configuration.py]: There is no suitable exception. I think it is better to create a custom one.

**Pull request merged**